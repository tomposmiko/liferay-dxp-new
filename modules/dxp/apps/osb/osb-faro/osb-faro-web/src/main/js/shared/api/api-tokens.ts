import sendRequest from 'shared/util/request';
import {RESTParams} from 'shared/types';

export function generate({groupId}: RESTParams) {
	return sendRequest({
		method: 'POST',
		path: `main/${groupId}/oauth2/tokens/new`
	});
}

export function search({groupId}: RESTParams) {
	return sendRequest({
		method: 'GET',
		path: `main/${groupId}/oauth2/tokens`
	});
}

export function revoke({groupId, token}: RESTParams & {token: string}) {
	return sendRequest({
		method: 'POST',
		path: `main/${groupId}/oauth2/tokens/${token}/revoke`
	});
}
