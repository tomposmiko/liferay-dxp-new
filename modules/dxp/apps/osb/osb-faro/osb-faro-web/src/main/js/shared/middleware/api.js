import sendRequest from 'shared/util/request';
import {get, isNil} from 'lodash';

export const CALL_API = 'CALL_API';

export function toAction(type, ...objs) {
	const action = Object.assign(...objs, {type});

	delete action.meta[CALL_API];

	return action;
}

export default () => next => action => {
	const request = get(action, ['meta', CALL_API]);

	if (isNil(request)) {
		return next(action);
	}

	const {data, requestFn, types} = request;

	const [requestType, successType, failureType] = types;

	next(toAction(requestType, action));

	const retVal = requestFn ? requestFn(data) : sendRequest(request);

	return retVal.then(
		payload => {
			next(
				toAction(successType, action, {
					meta: {
						...action.meta,
						schema: request.schema
					},
					payload
				})
			);

			return {payload};
		},
		error => {
			next(
				toAction(failureType, action, {
					error: true
				})
			);

			throw error;
		}
	);
};
