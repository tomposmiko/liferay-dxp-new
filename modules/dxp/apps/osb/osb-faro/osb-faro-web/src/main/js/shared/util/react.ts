import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
