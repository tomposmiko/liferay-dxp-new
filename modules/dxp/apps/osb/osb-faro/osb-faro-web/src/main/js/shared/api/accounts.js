import FaroConstants from 'shared/util/constants';
import sendRequest from 'shared/util/request';
import {ACCOUNTS} from 'shared/util/router';
import {buildOrderByFields} from 'shared/util/pagination';
import {escapeSingleQuotes} from 'segment/segment-editor/dynamic/utils/odata';

const {
	pagination: {cur: DEFAULT_PAGE, delta: DEFAULT_DELTA}
} = FaroConstants;

export function fetch({accountId, groupId}) {
	return sendRequest({
		method: 'GET',
		path: `contacts/${groupId}/account/${accountId}`
	});
}

export function fetchDetails({accountId, groupId}) {
	return sendRequest({
		method: 'GET',
		path: `contacts/${groupId}/account/${accountId}/details`
	});
}

export function fetchFieldValues({fieldMappingFieldName, groupId, query}) {
	return sendRequest({
		data: {
			delta: DEFAULT_DELTA,
			fieldMappingFieldName,
			query: escapeSingleQuotes(query)
		},
		method: 'GET',
		path: `contacts/${groupId}/account/field_values`
	});
}

export function search({
	channelId = '',
	delta = DEFAULT_DELTA,
	groupId,
	orderIOMap,
	page = DEFAULT_PAGE,
	query = '',
	...otherParams
}) {
	const orderParams = orderIOMap.first();

	const orderByFields = buildOrderByFields(orderParams, ACCOUNTS);

	return sendRequest({
		data: {
			channelId,
			cur: page,
			delta,
			orderByFields,
			query,
			...otherParams
		},
		method: 'POST',
		path: `contacts/${groupId}/account/search`
	});
}
