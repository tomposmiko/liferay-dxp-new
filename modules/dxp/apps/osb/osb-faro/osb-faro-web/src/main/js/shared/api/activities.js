import sendRequest from 'shared/util/request';
import {buildOrderByFields} from 'shared/util/pagination';
import {pickBy} from 'lodash';
import {TimeIntervals} from 'shared/util/constants';

export const DEFAULT_ACTIVITY_MAX = 30;

export const DEFAULT_ACTIVITY_INTERVAL = TimeIntervals.Day;

export function fetchHistory({
	channelId = '',
	contactsEntityId,
	contactsEntityType,
	groupId,
	interval = DEFAULT_ACTIVITY_INTERVAL,
	max = DEFAULT_ACTIVITY_MAX,
	rangeEnd = null,
	rangeStart = null
}) {
	return sendRequest({
		data: {
			channelId,
			contactsEntityId,
			contactsEntityType,
			interval,
			...pickBy({
				max,
				rangeEnd,
				rangeStart
			})
		},
		method: 'GET',
		path: `contacts/${groupId}/activity/history`
	});
}

export function fetchGroup({
	channelId = '',
	contactsEntityId,
	contactsEntityType,
	cur,
	delta,
	endDate,
	groupId,
	orderIOMap,
	query,
	startDate
}) {
	const orderParams = orderIOMap.first();

	const orderByFields = buildOrderByFields(orderParams);

	return sendRequest({
		data: {
			channelId,
			contactsEntityId,
			contactsEntityType,
			cur,
			delta,
			endDate,
			orderByFields,
			query,
			startDate
		},
		method: 'GET',
		path: `contacts/${groupId}/activity_group`
	});
}

export function searchAssets({
	applicationId,
	channelId = '',
	eventId,
	groupId,
	orderIOMap,
	...otherParams
}) {
	const orderParams = orderIOMap.first();

	const orderByFields = buildOrderByFields(orderParams);

	return sendRequest({
		data: {
			...otherParams,
			applicationId,
			channelId,
			eventId,
			orderByFields
		},
		method: 'GET',
		path: `contacts/${groupId}/activity/asset`
	});
}

export function searchCount({action, groupId, ...otherParams}) {
	return sendRequest({
		data: {
			...otherParams,
			action
		},
		method: 'GET',
		path: `contacts/${groupId}/activity/count`
	});
}
