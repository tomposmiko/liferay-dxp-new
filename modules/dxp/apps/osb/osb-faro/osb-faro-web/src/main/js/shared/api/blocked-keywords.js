import sendRequest from 'shared/util/request';
import {buildOrderByFields} from 'shared/util/pagination';

function delete$({groupId, ids}) {
	return sendRequest({
		data: {ids},
		method: 'DELETE',
		path: `main/${groupId}/blocked_keywords`
	});
}
export {delete$ as delete};

export function search({delta, groupId, orderIOMap, page, query}) {
	const orderParams = orderIOMap.first();

	const orderByFields = buildOrderByFields(orderParams);

	return sendRequest({
		data: {
			cur: page,
			delta,
			orderByFields,
			query
		},
		method: 'GET',
		path: `main/${groupId}/blocked_keywords`
	});
}

export function insertMany({groupId, keywords}) {
	return sendRequest({
		data: {keywords},
		method: 'POST',
		path: `main/${groupId}/blocked_keywords`
	});
}
