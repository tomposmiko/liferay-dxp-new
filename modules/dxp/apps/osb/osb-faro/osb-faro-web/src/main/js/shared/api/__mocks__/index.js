import * as accounts from './accounts';
import * as activities from './activities';
import * as apiTokens from './api-tokens';
import * as assets from './assets';
import * as blockedKeywords from './blocked-keywords';
import * as channels from './channels';
import * as contactsCards from './contacts-cards';
import * as dataSource from './data-source';
import * as definitions from './definitions';
import * as distributions from './distributions';
import * as fieldMappings from './field-mappings';
import * as individuals from './individuals';
import * as individualSegment from './individual-segment';
import * as interests from './interests';
import * as issue from './issue';
import * as notifications from './notifications';
import * as pagesVisited from './pages-visited';
import * as preferences from './preferences';
import * as projects from './projects';
import * as user from './user';

export {
	accounts,
	activities,
	apiTokens,
	assets,
	blockedKeywords,
	channels,
	contactsCards,
	dataSource,
	definitions,
	distributions,
	fieldMappings,
	individualSegment,
	individuals,
	interests,
	issue,
	notifications,
	pagesVisited,
	preferences,
	projects,
	user
};
