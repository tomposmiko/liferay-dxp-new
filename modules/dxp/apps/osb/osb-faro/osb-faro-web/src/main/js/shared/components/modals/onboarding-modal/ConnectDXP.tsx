import * as API from 'shared/api';
import BaseScreen from './BaseScreen';
import Button from 'shared/components/Button';
import CopyButton from 'shared/components/CopyButton';
import DataSourceQuery, {
	DataSource,
	DataSourceData,
	DataSourceSyncData
} from 'shared/queries/DataSourceQuery';
import getCN from 'classnames';
import Icon from 'shared/components/Icon';
import InfoPopover from 'shared/components/InfoPopover';
import Input from 'shared/components/Input';
import Label from 'shared/components/form/Label';
import Modal from 'shared/components/modal';
import React, {FC, useEffect, useRef, useState} from 'react';
import Select from 'shared/components/Select';
import StampLabel from 'shared/components/Label';
import URLConstants from 'shared/util/url-constants';
import {ActionType, useChannelContext} from 'shared/context/channel';
import {compose} from 'redux';
import {connect} from 'react-redux';
import {CREATE_DATE} from 'shared/util/pagination';
import {
	CredentialTypes,
	DataSourceTypes,
	OrderByDirections
} from 'shared/util/constants';
import {fetchDataSource} from 'shared/actions/data-sources';
import {get, noop, upperFirst} from 'lodash';
import {getDefaultChannel} from 'shared/components/channels-menu';
import {Routes, toRoute} from 'shared/util/router';
import {sub} from 'shared/util/lang';
import {useInterval} from 'shared/hooks/useInterval';
import {useLazyQuery} from '@apollo/react-hooks';
import {withHistory} from 'shared/hoc';

const TIMEOUT_INTERVAL = 5000;

const DXP_VERSIONS = {
	'dxp-70-fixpack-98': {
		label: 'DXP 7.0 Fix Pack 98',
		url: URLConstants.DownloadDXP70FixPack98
	},
	'dxp-71-fixpack-22': {
		label: 'DXP 7.1 Fix Pack 22',
		url: URLConstants.DownloadDXP71FixPack22
	},
	'dxp-72-fixpack-1': {
		label: 'DXP 7.2 Fix Pack 11',
		url: URLConstants.DownloadDXP72FixPack11
	},
	'dxp-73-fixpack-1': {
		label: 'DXP 7.3 Fix Pack 1',
		url: URLConstants.DownloadDXP73FixPack1
	}
};

const DATA_SOURCE_STATUSES = {
	CONFIGURED: {
		display: StampLabel.Displays.Success,
		label: Liferay.Language.get('configured')
	},
	UNCONFIGURED: {
		display: StampLabel.Displays.Secondary,
		label: Liferay.Language.get('unconfigured')
	}
};

interface IConnectDXPProps {
	dxpConnected: boolean;
	groupId: string;
	onboarding?: boolean;
	onClose: () => void;
	onNext?: (increment?: number) => void;
}

interface IConnectDXPWrapperProps {
	dataSourceId?: string;
	fetchDataSource: ({
		groupId,
		id
	}: {
		groupId: string;
		id: string;
	}) => DataSource;
	history: {
		push: (path: string) => void;
	};
	isUpgrading: boolean;
	onDxpConnected: (dxpConnected: boolean) => void;
	onPrevious?: () => void;
}

interface IFooterProps {
	data: DataSourceData;
}

interface ITokenInputProps {
	token: string;
}

const ConnectDXP: React.FC<IConnectDXPWrapperProps & IConnectDXPProps> = ({
	dataSourceId,
	dxpConnected,
	fetchDataSource,
	groupId,
	history,
	onboarding,
	onClose,
	onDxpConnected,
	onNext
}) => {
	const {channelDispatch} = useChannelContext();
	const [token, setToken] = useState<string>('');

	const [getDataSources, {data}] = useLazyQuery<DataSourceData>(
		DataSourceQuery,
		{
			fetchPolicy: 'network-only',
			onCompleted: () => {
				onDxpConnected(true);
			},
			variables: {
				credentialsType: CredentialTypes.Token,
				size: 1,
				sort: {
					column: CREATE_DATE,
					type: OrderByDirections.Descending
				},
				type: DataSourceTypes.Liferay
			}
		}
	);

	let _tokenRequest;

	const getNextToken: (prevToken?: string) => Promise<any> = prevToken =>
		API.dataSource
			.fetchToken(groupId, dataSourceId)
			.then(nextToken => {
				if (!prevToken || prevToken === nextToken) {
					_tokenRequest = setTimeout(
						() => getNextToken(nextToken),
						TIMEOUT_INTERVAL
					);
				} else {
					analytics.track(
						'Established connection w/ DXP - TEST',
						null,
						{
							ip: '0'
						}
					);

					if (onboarding) {
						onDxpConnected(true);

						updateChannels();
						// if it's an upgrade from oauth to token, we need to fetch the DataSource
					} else if (dataSourceId) {
						fetchDataSource({groupId, id: dataSourceId});
					}

					getDataSources();
				}

				return nextToken;
			})
			.catch(err => {
				if (!err.IS_CANCELLATION_ERROR) {
					_tokenRequest = setTimeout(
						() => getNextToken(prevToken),
						TIMEOUT_INTERVAL
					);
				}

				return prevToken;
			});

	const updateChannels = () => {
		API.channels.fetchAll({groupId}).then(({items}) => {
			const channelId = get(items, [0, 'id']);

			history.push(toRoute(Routes.SITES, {channelId, groupId}));

			channelDispatch({
				payload: getDefaultChannel(channelId, items),
				type: ActionType.setSelectedChannel
			});

			channelDispatch({
				payload: items,
				type: ActionType.setChannels
			});
		});
	};

	useEffect(() => {
		_tokenRequest = getNextToken().then(setToken);

		return () => {
			clearTimeout(_tokenRequest);
		};
	}, []);

	return (
		<BaseScreen className='connect-dxp' onClose={onClose}>
			<Modal.Body className='d-flex flex-column align-items-center flex-grow-1 justify-content-center'>
				<div className='analytics-to-dxp-container'>
					<Icon size='xl' symbol='dxp-icon' />

					<Icon
						className={getCN('arrows', {connected: dxpConnected})}
						size='lg'
						symbol='ac-horizontal-arrows'
					/>

					<Icon
						size='xl'
						symbol={dxpConnected ? 'ac-logo' : 'ac-logo-grayscale'}
					/>
				</div>

				<span className='title connect-to-dxp d-flex justify-content-center'>
					{dxpConnected
						? Liferay.Language.get(
								'your-dxp-instance-is-connected-to-analytics-cloud'
						  )
						: Liferay.Language.get('connect-your-dxp-analytics')}
				</span>

				{!dxpConnected && (
					<>
						<TokenInput token={token} />

						<FixPackSelect />
					</>
				)}

				{dxpConnected && <DxpSyncTable />}
			</Modal.Body>

			<Footer
				data={data}
				dxpConnected={dxpConnected}
				groupId={groupId}
				onboarding={onboarding}
				onClose={onClose}
				onNext={onNext}
			/>
		</BaseScreen>
	);
};

const DxpSyncTable: FC<React.HTMLAttributes<HTMLElement>> = () => {
	const [dataSource, setDataSources] = useState<DataSource>({
		contactsSyncDetails: {selected: false},
		sitesSyncDetails: {selected: false}
	});
	const [getDataSources, {data}] = useLazyQuery<DataSourceSyncData>(
		DataSourceQuery,
		{
			fetchPolicy: 'network-only',
			variables: {
				credentialsType: CredentialTypes.Token,
				size: 1,
				sort: {
					column: CREATE_DATE,
					type: OrderByDirections.Descending
				},
				type: DataSourceTypes.Liferay
			}
		}
	);
	useInterval<void>(getDataSources, TIMEOUT_INTERVAL);

	const getLabelProps = (selected: boolean) =>
		selected
			? DATA_SOURCE_STATUSES.CONFIGURED
			: DATA_SOURCE_STATUSES.UNCONFIGURED;

	const {display: contactsDisplay, label: contactslabel} = getLabelProps(
		dataSource?.contactsSyncDetails?.selected
	);

	const {display: sitesDisplay, label: sitesLabel} = getLabelProps(
		dataSource?.sitesSyncDetails?.selected
	);

	useEffect(() => {
		if (data) {
			setDataSources(data.dataSources[0]);
		}
	}, [data]);

	return (
		<div className='success-info w-100'>
			<div className='container'>
				<div className='row'>
					<span className='col'>{Liferay.Language.get('sites')}</span>

					<div className='col'>
						<StampLabel display={sitesDisplay}>
							{sitesLabel}
						</StampLabel>
					</div>
				</div>

				<div className='row'>
					<span className='col'>
						{Liferay.Language.get('contacts')}
					</span>

					<div className='col'>
						<StampLabel display={contactsDisplay}>
							{contactslabel}
						</StampLabel>
					</div>
				</div>
			</div>

			<div className='secondary-description mt-1 w-100'>
				{Liferay.Language.get(
					'you-can-check-your-data-source-syncing-status-under-settings-in-data-sources,-or-continue-configuring-your-data-source-on-your-dxp-instance'
				)}
			</div>
		</div>
	);
};

const Footer: FC<IFooterProps & IConnectDXPProps> = ({
	data,
	dxpConnected,
	groupId,
	onboarding,
	onClose,
	onNext
}) => {
	const getNavHref = () => {
		const id = get(data, ['dataSources', 0, 'id'], null);

		if (id) {
			return toRoute(Routes.SETTINGS_DATA_SOURCE, {groupId, id});
		}

		return toRoute(Routes.SETTINGS_DATA_SOURCE_LIST, {groupId});
	};

	return (
		<Modal.Footer className='d-flex justify-content-end'>
			<div>
				{!(dxpConnected && onboarding) && (
					<Button
						disabled={dxpConnected}
						onClick={onboarding ? () => onNext() : onClose}
					>
						{onboarding
							? Liferay.Language.get('skip')
							: Liferay.Language.get('cancel')}
					</Button>
				)}

				<Button
					disabled={!dxpConnected}
					display='primary'
					href={onboarding || !dxpConnected ? null : getNavHref()}
					onClick={onboarding ? () => onNext() : onClose}
				>
					{onboarding
						? Liferay.Language.get('next')
						: Liferay.Language.get('done')}
				</Button>
			</div>
		</Modal.Footer>
	);
};

const FixPackSelect: FC<React.HTMLAttributes<HTMLElement>> = () => {
	const dxpVersionsList = Object.keys(DXP_VERSIONS);
	const [dxpVersion, setDxpVersion] = useState<string>(dxpVersionsList[0]);

	return (
		<>
			<div className='secondary-description w-100 ml-6 mt-1'>
				{sub(
					Liferay.Language.get(
						'x-to-learn-how-to-connect-liferay-dxp-to-analytics-cloud'
					),
					[
						<a
							href={URLConstants.HelpConnectDxp}
							key='helpConnectDxpText'
							target='_blank'
						>
							{upperFirst(
								Liferay.Language.get('click-here').toLowerCase()
							)}
						</a>
					],
					false
				)}
			</div>

			<div className='fix-pack-container'>
				<div className='fix-pack-select'>
					<Label>
						{Liferay.Language.get('dxp-fix-pack-requirements')}

						<InfoPopover
							className='ml-2'
							content={Liferay.Language.get(
								'minimum-fix-pack-version-required-for-full-functionality'
							)}
							title={Liferay.Language.get(
								'dxp-fix-pack-requirements'
							)}
						/>
					</Label>
					<Select
						className='mt-1'
						onChange={({target: {value}}) => setDxpVersion(value)}
						value={dxpVersion}
					>
						{dxpVersionsList.map(key => (
							<Select.Item key={key} value={key}>
								{DXP_VERSIONS[key].label}
							</Select.Item>
						))}
					</Select>
				</div>

				<div className='fix-pack-button'>
					<Button
						borderless
						className='more-information-link mt-4'
						externalLink
						href={DXP_VERSIONS[dxpVersion].url}
						icon='shortcut'
						iconAlignment='right'
						target='_blank'
					>
						{Liferay.Language.get('download')}
					</Button>
				</div>
			</div>
		</>
	);
};

const TokenInput: FC<ITokenInputProps> = ({token}) => {
	const [tokenCopied, setTokenCopied] = useState(false);
	const copyButtonClassName = getCN('copy-button', {
		'input-success': tokenCopied
	});

	const _inputRef = useRef<any>();
	const selectAll = () => {
		analytics.track('Clicked Copy Token Button - TEST', null, {
			ip: '0'
		});

		_inputRef.current && _inputRef.current.selectAll();
	};

	return (
		<>
			<div className='w-100 ml-6 mt-1 token-info'>
				{Liferay.Language.get('copy-this-token-to-your-dxp-instance')}
			</div>

			<Input.Group>
				<Input.GroupItem position='prepend'>
					<Input
						className={getCN('text-truncate', 'token-input', {
							'input-success': tokenCopied
						})}
						inset='after'
						onChange={noop}
						onClick={selectAll}
						ref={_inputRef}
						value={token}
					/>

					<Input.Inset
						className={copyButtonClassName}
						position='after'
					>
						<CopyButton
							className={copyButtonClassName}
							display='light'
							onClick={() => {
								setTokenCopied(true);

								analytics.track(
									'Clicked Copy Token Button - TEST',
									null,
									{ip: '0'}
								);
							}}
							text={token}
						/>
					</Input.Inset>
				</Input.GroupItem>
			</Input.Group>
		</>
	);
};

export default compose<any>(
	withHistory,
	connect(null, {
		fetchDataSource
	})
)(ConnectDXP);
