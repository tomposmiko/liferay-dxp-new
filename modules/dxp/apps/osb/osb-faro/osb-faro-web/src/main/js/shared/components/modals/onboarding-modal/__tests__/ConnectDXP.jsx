import * as API from 'shared/api';
import ConnectDXP from '../ConnectDXP';
import mockStore from 'test/mock-store';
import Promise from 'metal-promise';
import React from 'react';
import {fireEvent, render} from '@testing-library/react';
import {noop} from 'lodash';
import {Provider} from 'react-redux';
import {StaticRouter} from 'react-router-dom';

jest.unmock('react-dom');

describe('ConnectDXP', () => {
	it('renders', () => {
		const {container} = render(
			<Provider store={mockStore()}>
				<StaticRouter>
					<ConnectDXP groupId='123' onClose={noop} onNext={noop} />
				</StaticRouter>
			</Provider>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders "Connected" when dxpConnected is true', () => {
		const {queryByText} = render(
			<Provider store={mockStore()}>
				<StaticRouter>
					<ConnectDXP
						dxpConnected
						groupId='123'
						onClose={noop}
						onNext={noop}
					/>
				</StaticRouter>
			</Provider>
		);

		expect(queryByText('Back')).toBeNull();
		expect(
			queryByText('Your DXP Instance Is Connected to Analytics Cloud')
		).toBeInTheDocument();
		expect(queryByText('Sites')).toBeInTheDocument();
		expect(queryByText('Contacts')).toBeInTheDocument();
	});

	it('renders Download button', () => {
		const {queryByText} = render(
			<Provider store={mockStore()}>
				<StaticRouter>
					<ConnectDXP groupId='123' onClose={noop} onNext={noop} />
				</StaticRouter>
			</Provider>
		);

		expect(queryByText('Download')).toBeTruthy();
	});

	it('change Download link when change DXP version', () => {
		const {container, queryByText} = render(
			<Provider store={mockStore()}>
				<StaticRouter>
					<ConnectDXP groupId='123' onClose={noop} onNext={noop} />
				</StaticRouter>
			</Provider>
		);

		const select = container.querySelector('.select-root');

		fireEvent.change(select, {
			target: {value: 'dxp-71-fixpack-22'}
		});

		expect(queryByText('Download').href).toMatch(/7-1-fix-pack-22/);
	});

	it('fires "setDxpConnected" when the token value changes', () => {
		const spy = jest.fn();

		render(
			<Provider store={mockStore()}>
				<StaticRouter>
					<ConnectDXP
						groupId='123'
						onboarding
						onClose={noop}
						onDxpConnected={spy}
						onNext={noop}
					/>
				</StaticRouter>
			</Provider>
		);

		expect(spy).not.toBeCalled();

		jest.runOnlyPendingTimers();

		API.dataSource.fetchToken.mockReturnValue(
			Promise.resolve('New Token Value')
		);

		jest.runOnlyPendingTimers();
		jest.runOnlyPendingTimers();

		expect(spy).toBeCalled();
	});
});
