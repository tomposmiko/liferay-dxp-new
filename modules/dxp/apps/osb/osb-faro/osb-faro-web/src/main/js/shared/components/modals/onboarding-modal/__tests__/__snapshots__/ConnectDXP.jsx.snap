// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectDXP renders 1`] = `
<div>
  <div
    class="onboarding-base-screen-root d-flex flex-column flex-grow-1 justify-content-between connect-dxp"
  >
    <div
      class="header"
    >
      <div
        class="d-flex justify-content-end"
      >
        <button
          class="button-root close btn btn-secondary"
          type="button"
        >
          <svg
            class="lexicon-icon lexicon-icon-times icon-root"
            role="presentation"
          >
            <use
              xlink:href="#times"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="modal-body d-flex flex-column align-items-center flex-grow-1 justify-content-center"
    >
      <div
        class="analytics-to-dxp-container"
      >
        <svg
          class="lexicon-icon lexicon-icon-dxp-icon icon-root icon-size-xl"
          role="presentation"
        >
          <use
            xlink:href="#dxp-icon"
          />
        </svg>
        <svg
          class="lexicon-icon lexicon-icon-ac-horizontal-arrows arrows icon-root icon-size-lg"
          role="presentation"
        >
          <use
            xlink:href="#ac-horizontal-arrows"
          />
        </svg>
        <svg
          class="lexicon-icon lexicon-icon-ac-logo-grayscale icon-root icon-size-xl"
          role="presentation"
        >
          <use
            xlink:href="#ac-logo-grayscale"
          />
        </svg>
      </div>
      <span
        class="title connect-to-dxp d-flex justify-content-center"
      >
        Connect Your DXP Analytics
      </span>
      <div
        class="w-100 ml-6 mt-1 token-info"
      >
        Copy this token to your DXP instance.
      </div>
      <div
        class="input-group"
      >
        <div
          class="input-group-item input-group-prepend"
        >
          <input
            class="input-root form-control text-truncate token-input input-group-inset input-group-inset-after"
            type="text"
            value=""
          />
          <div
            class="input-group-inset-item copy-button input-group-inset-item-after"
          >
            <button
              aria-label="Click to Copy"
              class="copy-button btn btn-unstyled"
              data-clipboard-text=""
              data-tooltip-response="Copied"
              title="Click to Copy"
              type="button"
            >
              <svg
                class="lexicon-icon lexicon-icon-paste icon-root"
                role="presentation"
              >
                <use
                  xlink:href="#paste"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="secondary-description w-100 ml-6 mt-1"
      >
        <a
          href="https://learn.liferay.com/analytics-cloud/latest/en/connecting-data-sources/connecting-liferay-dxp-to-analytics-cloud.html"
          target="_blank"
        >
          Click here
        </a>
         to learn how to connect Liferay DXP to Analytics Cloud.
      </div>
      <div
        class="fix-pack-container"
      >
        <div
          class="fix-pack-select"
        >
          <label
            class="form-control-label label-root"
          >
            <span
              class="content-container"
            >
              DXP Fix Pack Requirements
              <span
                class="info-popover-root ml-2"
              >
                <svg
                  class="lexicon-icon lexicon-icon-question-circle-full icon-root"
                  role="presentation"
                >
                  <use
                    xlink:href="#question-circle-full"
                  />
                </svg>
              </span>
            </span>
          </label>
          <select
            class="form-control select-root mt-1"
          >
            <option
              value="dxp-70-fixpack-98"
            >
              DXP 7.0 Fix Pack 98
            </option>
            <option
              value="dxp-71-fixpack-22"
            >
              DXP 7.1 Fix Pack 22
            </option>
            <option
              value="dxp-72-fixpack-1"
            >
              DXP 7.2 Fix Pack 11
            </option>
            <option
              value="dxp-73-fixpack-1"
            >
              DXP 7.3 Fix Pack 1
            </option>
          </select>
        </div>
        <div
          class="fix-pack-button"
        >
          <a
            class="button-root more-information-link mt-4 btn btn-secondary"
            href="https://customer.liferay.com/downloads/-/download/liferay-dxp-7-0-fix-pack-98"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              class="lexicon-icon lexicon-icon-shortcut icon-root mr-2"
              role="presentation"
            >
              <use
                xlink:href="#shortcut"
              />
            </svg>
            Download
            <span
              class="sr-only"
            >
              (Opens a new window)
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="modal-footer d-flex justify-content-end"
    >
      <div>
        <button
          class="button-root btn btn-secondary"
          type="button"
        >
          Cancel
        </button>
        <button
          class="button-root btn btn-primary"
          disabled=""
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;
